hack/update-codecgen.sh

```diff
@@ -41,6 +41,46 @@ function cleanup {
 }
 trap cleanup EXIT
 
+# Sort all files in the dependency order.
+number=${#generated_files[@]}
+for (( i=0; i<number; i++ )); do
+  visited[${i}]=false
+done
+result=""
+
+function depends {
+  file=${generated_files[$1]//\.generated\.go/.go}
+  deps=$(go list -f "{{.Deps}}" ${file} | tr "[" " " | tr "]" " ")
+  fullpath=$(readlink -f ${generated_files[$2]//\.generated\.go/.go})
+  candidate=$(dirname "${fullpath}")
+  result=false
+  for dep in ${deps}; do
+    if [[ ${candidate} = *${dep} ]]; then
+      result=true
+    fi
+  done
+  echo ${result}
+}
+
+function tsort {
+  visited[$1]=true
+  local j=0
+  for (( j=0; j<number; j++ )); do
+    if ! ${visited[${j}]}; then
+      if $(depends "$1" ${j}); then
```

Comment:
Do you need the `$(...)` here? I think maybe you can just call a function in an if (I'm no bash expert though).